research/deep_speech/data/download.py

```diff
@@ -0,0 +1,171 @@
+#  Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ==============================================================================
+"""Download and preprocess LibriSpeech dataset for DeepSpeech model."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import codecs
+import fnmatch
+import os
+import tarfile
+import tempfile
+import unicodedata
+
+from absl import app as absl_app
+from absl import flags as absl_flags
+import pandas
+from six.moves import urllib
+from sox import Transformer
+import tensorflow as tf
+
+LIBRI_SPEECH_URLS = {
+    "train-clean-100":
+        "http://www.openslr.org/resources/12/train-clean-100.tar.gz",
+    "train-clean-360":
+        "http://www.openslr.org/resources/12/train-clean-360.tar.gz",
+    "train-other-500":
+        "http://www.openslr.org/resources/12/train-other-500.tar.gz",
+    "dev-clean":
+        "http://www.openslr.org/resources/12/dev-clean.tar.gz",
+    "dev-other":
+        "http://www.openslr.org/resources/12/dev-other.tar.gz",
+    "test-clean":
+        "http://www.openslr.org/resources/12/test-clean.tar.gz",
+    "test-other":
+        "http://www.openslr.org/resources/12/test-other.tar.gz"
+}
+
+
+def download_and_extract(directory, url):
+  """Download and extract the given split of dataset.
+
+  Args:
+    directory: the directory where to extract the tarball.
+    url: the url to download the data file.
+  """
+
+  if not tf.gfile.Exists(directory):
+    tf.gfile.MakeDirs(directory)
+
+  _, tar_filepath = tempfile.mkstemp(suffix=".tar.gz")
+  try:
+    tf.logging.info("Downloading %s to %s" % (url, tar_filepath))
+    urllib.request.urlretrieve(url, tar_filepath)
```

Comment:
Could you add a progress bar to track the downloading process? As it takes long for data download, and it just freezes there with no cues. A good reference is cifar10 download: https://github.com/tensorflow/models/blob/add588d68888e8ba869ffce17d214c48e41ca019/official/resnet/cifar10_download_and_extract.py#L48